x-pack/plugins/stack_alerts/server/alert_types/es_query/action_context.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { AlertExecutorOptions, AlertInstanceContext } from '../../../../alerts/server';
+import { EsQueryAlertParams } from './alert_type_params';
+
+// alert type context provided to actions
+
+type AlertInfo = Pick<AlertExecutorOptions, 'name'>;
+
+export interface ActionContext extends EsQueryAlertActionContext {
+  // a short pre-constructed message which may be used in an action field
+  title: string;
+  // a longer pre-constructed message which may be used in an action field
+  message: string;
+}
+
+export interface EsQueryAlertActionContext extends AlertInstanceContext {
+  // the date the alert was run as an ISO date
+  date: string;
+  // the value that met the threshold
+  value: number;
+  // threshold conditions
+  conditions: string;
+  // query matches
+  hits: unknown[];
```

Comment:
Can we use the ES typing for hits here? 🤔  Or is this something else?

Reply:
Good call! Updated.